core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/TypeUtils.kt

```diff
@@ -436,6 +436,8 @@ internal fun guessValueType(values: Sequence<Any?>, upperBound: KType? = null, l
                 collectionClasses.add(it.javaClass.kotlin)
             }
 
+            is Function<*> -> classes.add(Function::class)
```

Comment:
why not :) Maybe we should change the rendering for functions in dataframes though. After a quick test I found it looks like:
```
⌌---------------------------------------------------------------------------------------------------------------⌍
|  |                                                                                        a:Function<*>| b:Int|
|--|-----------------------------------------------------------------------------------------------------|------|
| 0| org.jetbrains.kotlinx.dataframe.testSets.person.DataFrameTests$$Lambda$60/0x000000010013a040@64ee819|     2|
⌎---------------------------------------------------------------------------------------------------------------⌏
```

Reply:
Sadly for such lambda objects toString is weird. I tried to look at the object in the debugger, but there's literally nothing that hints at signature or anything useful

Reply:
hmm you'd think there was a way in kotlin to detect it's a `() -> Int` or something :/

Reply:
actually...

![image](https://github.com/user-attachments/assets/0fe995d3-fc78-44fb-92bf-49a96c6882e2)

Reply:
It already renders correctly often!

![image](https://github.com/user-attachments/assets/1fcdd7cf-6192-4c9f-94a7-b25a594f66e7)

might just be a fluke in the tests if the lambda is serialized as interface

Reply:
what kernel version do you use?
![image](https://github.com/user-attachments/assets/efa600a2-8d3c-4a1d-839a-3991ff11819f)

Reply:
I run the dev version of this PR's branch in the notebook

Reply:
Ah, ok, so the fix is needed anyway